lib/kafo/param.rb

```diff
@@ -34,8 +35,34 @@ def unset_value
       @value     = nil
     end
 
+    # For literal default values, only use 'manifest_default'. For variable values, use the value
+    # loaded back from the dump in 'default'.
+    def default
+      @type.typecast(dump_default_needed? ? @default : manifest_default)
+    end
+
+    def default=(default)
+      default = nil if ['UNSET', :undef].include?(default)
```

Comment:
could you extract `['UNSET', :undef]` from here and `manifest_default=` to some constant, e.g. UNSETTING_VALUES so they do not get out of sync?